### PR TITLE
New: CountBadge component

### PR DESCRIPTION
**Pull request by nimishjha**
## Description

A react component for displaying numerical count in a fixed size badge.

## Motivation and Background Context

Needed to show warning and error counts as well as tallies of information in various parts of the platform. See https://github.com/Adslot/adslot-ui/issues/729

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Tested on the examples page. As it's a new component, it doesn't affect any existing components. 

## Screenshots (if appropriate):
countbadgeexample

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [ ] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [x] I've two LGTMs/Approvals.
- [x] I've fixed or replied to all my code-review comments.
- [x] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: docs/components/Layout/index.jsx [hunk 1]
```diff
@@ -83,7 +84,7 @@ const componentsBySection = {
     'date-picker',
   ],
   'typography-and-text-layout': ['text-ellipsis'],
-  'stats-and-data': ['statistic', 'totals', 'slicey'],
+  'stats-and-data': ['count-badge', 'statistic', 'totals', 'slicey'],
```
Comment (omgaz) [inline docs/components/Layout/index.jsx]: 👍

File: src/components/adslot-ui/CountBadge/styles.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  &.status-warning {
+    background-color: $color-warning;
+  }
+
+  &.status-danger {
+    background-color: $color-negative;
+  }
+
+  &.font-size-small {
```
Comment (omgaz) [inline src/components/adslot-ui/CountBadge/styles.scss]: Just worried about direct web or cowboy css abuse out there in third-party libraries.

Do you think we should protect with:

`.count-badge-font-size-small`

File: docs/examples/CountBadgeExample.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+/>`,
+  propTypeSectionArray: [
+    {
+      label: '',
+      propTypes: [
+        {
+          propType: '',
+          type: '',
+          defaultValue: '',
+          note: '',
```
[the remark this answers is not in the document]
Comment (nimishjha) [reply docs/examples/CountBadgeExample.jsx]: Oops

File: src/components/adslot-ui/CountBadge/styles.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
+@import '~styles/variable';
+
+.count-badge {
+  display: inline-block;
+  width: 16px;
+  height: 16px;
```
Comment (omgaz) [inline src/components/adslot-ui/CountBadge/styles.scss]: Either use `$svg-icon-size` or create a new one scoped to `.count-badge`?

```
.count-badge {
  $size: 16px;

  width: $size;
  height: $size;
  …
}
```

Comment (nimishjha) [reply src/components/adslot-ui/CountBadge/styles.scss]: 👍

File: src/components/adslot-ui/CountBadge/styles.scss (hunk 2 above)
Comment (nimishjha) [reply src/components/adslot-ui/CountBadge/styles.scss]: 👍

File: src/components/adslot-ui/CountBadge/index.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    return (
+      <div className={classNames} {...expandDts(dts)}>
+        <div className="count-badge-value">{value}</div>
+      </div>
+    );
+  }
+}
+
+CountBadge.propTypes = {
+  value: PropTypes.number,
```
Comment (xiaofan2406) [inline src/components/adslot-ui/CountBadge/index.jsx]: `isRequried`?

Comment (nimishjha) [reply src/components/adslot-ui/CountBadge/index.jsx]: 👍

File: src/components/adslot-ui/CountBadge/styles.scss [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  &.status-warning {
+    background-color: $color-warning;
+  }
+
+  &.status-danger {
+    background-color: $color-negative;
+  }
+
+  &.count-badge-font-size-small {
+    font-size: $font-size-tiny;
```
Comment (omgaz) [inline src/components/adslot-ui/CountBadge/styles.scss]: should be 8px

File: src/components/adslot-ui/CountBadge/styles.scss [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  display: inline-block;
+  width: $svg-icon-size;
+  height: $svg-icon-size;
+  border-radius: 50%;
+  border-width: 1px;
+  border-color: $color-white;
+  background-color: $color-gray-dark;
+  color: $color-white;
+  text-align: center;
+  font-size: $font-size-small;
```
Comment (omgaz) [inline src/components/adslot-ui/CountBadge/styles.scss]: should be 9px

File: src/components/adslot-ui/CountBadge/styles.scss (hunk 6 above)
Comment (nimishjha) [reply src/components/adslot-ui/CountBadge/styles.scss]: What should I call the new variable, $font-size-super-tiny?

**Comment by omgaz** [comment]
Can we also do something about reducing the DOM size from a count-badge + count-value to just one node as we're expecting upwards of 500-1000 of these on a page. Should eb able to do this by removing the inner value and adding a `line-height: 16px;`

**Pull request merged**